application/controllers/Admin.php

```diff
@@ -55,6 +56,87 @@ public function markAsPaid() {
         }
     }
 
+    /**
+     * Checks an input key against a key stored in a file. If it matches, store a cookie on the users browser.
+     */
+    public function authenticate() {
+
+        //gets the passkey from private_keys/auth_props.json
+        if (!defined('Passkey')) {
+            $json = file_get_contents("private_keys/auth_props.json");
+            $authProps = json_decode($json, true);
+        
+            define('Passkey', $authProps["auth.passkey"]);
+        }
+        // Scrambler key
+        $jwtKey = 'JWTKEY';
+
+        // Gets a key from the URL from the form admin/authenticate?key=xyz
+        $urlKey = $this->input->get('key');
+        
+        // Check if it matches a key we have stored
+        if ($urlKey != Passkey) {
+            echo("Key is incorrect");
+            return false;
+        }
+
+        $cookieName = 'admin_authentication';
+
+        $payload = array(
+            "key" => $urlKey,
+            "iat" => microtime(),
+        );
+
+        $jwt = JWT::encode($payload, $jwtKey);
+
+        setcookie($cookieName , $jwt); 
+        
+        echo 'Cookie set';
+        return true;
+    }
+
+
+    /**
+     * Check if a user has a specific cookie, and if they do allow them to do different things.
+     */
+    public function checkCookie() {
```

Comment:
Define this as a private method, since this will likely be used in other methods (inside this admin class). Also, remove all the `echo` in this method since this will be called by internal methods. Maybe do the echo's last since they're still useful for testing purposes?